acstools/findsat_mrt.py

```diff
@@ -309,44 +271,265 @@ def __init__(
+        due to diffraction spikes that always create streaks around the
+        same angle for a given telescope/instrument. Format should be a
+        list of tuples, e.g., ``[(theta0_a, theta1_a), (theta0_b, theta1_b)]``.
+        """
+        return self._ignore_theta_range
+
+    @ignore_theta_range.setter
+    def ignore_theta_range(self, value):
+        if ((value is None) or
+                (isinstance(value, list) and all(len(x) == 2 for x in value))):
+            self._ignore_theta_range = value
+        else:
+            raise ValueError(f"Invalid ignore_theta_range: {value}")
+
+    @property
+    def save_catalog(self):
+        """Save the catalog of identified trails to a FITS table."""
+        return self._save_catalog
+
+    @save_catalog.setter
+    def save_catalog(self, value):
+        if not isinstance(value, bool):
+            raise ValueError(f"save_catalog must be bool but got: {value}")
+        self._save_catalog = value
+
+    @property
+    def save_diagnostic(self):
+        """Save a diagnotic plot showing the input image and identified trails to a PNG file."""
+        return self._save_diagnostic
+
+    @save_diagnostic.setter
+    def save_diagnostic(self, value):
+        if not isinstance(value, bool):
+            raise ValueError(f"save_diagnostic must be bool but got: {value}")
+        self._save_diagnostic = value
+
+    @property
+    def save_mrt(self):
+        """Save the MRT in a FITS file."""
+        return self._save_mrt
+
+    @save_mrt.setter
+    def save_mrt(self, value):
+        if not isinstance(value, bool):
+            raise ValueError(f"save_mrt must be bool but got: {value}")
+        self._save_mrt = value
+
+    @property
+    def save_mask(self):
+        """Save the trail mask in a FITS file."""
+        return self._save_mask
+
+    @save_mask.setter
+    def save_mask(self, value):
+        if not isinstance(value, bool):
+            raise ValueError(f"save_mask must be bool but got: {value}")
+        self._save_mask = value
+
+    def run_mrt(self, plot=False):
         '''
-        Runs the median radon transform on the input image.
+        Run the median radon transform on the input image.
+        This uses `theta` and `processes` for calculations, so update them first, if needed.
 
         Parameters
         ----------
-        theta : array, optional
-            List of angles at which to calculate the MRT. The default is None,
-            which defers to self attribute of same name.
-        threads : int, optional
-            Number of CPU threads used to calculate the MRT. The default is
-            None, which defers to self attribute of same name.
         plot : bool, optional
-            Flag to plot the resulting MRT. The default is None, which defers
-            to the self attribute of same name.
-        Returns
-        -------
-        None.
+            Flag to plot the resulting MRT. The default is `False`.
```

Comment:
If TrailFinder.plot is True, the user is intending to plot wherever it's allowed, but plot is  False by default here, so no plots will be generated. Is there a way to have the plot keyword inherit the plot attribute from TrailFinder in cases where the plot keyword is not explicitly defined when calling run_mrt()?

Reply:
It is possible but I am unclear on what is the intended plotting behavior for this module. Even if it is False, one can still call the `plot_mrt()` method separately after (that is all it is doing here anyway), so no big deal.

It seems confusing to me to set plot as class attribute but then also let individual methods turn it on/off.

Should we just change plot to be class attribute and remove that option from methods? But then all your warnings about "this takes a lot of resources to plot" made me hesitant to do that.

Another option is not to have the method plot at all and let user call the separate plotting methods themselves, though that will not work if you are plotting intermediate results that will not be saved in the class.

Any preference?

Reply:
I've revised the code to remove the plot keyword from individual methods. Users can just update self.plot as needed.